src/main/java/joe/tasks/Task.java

```diff
@@ -0,0 +1,73 @@
+package joe.tasks;
+
+/**
+ * Represents a task in the task list.
+ */
+public abstract class Task {
+    private final String description;
+    private boolean isDone;
+
+    /**
+     * Constructor for Task.
+     *
+     * @param description a String describing the Task
+     * @throws IllegalArgumentException if description.isEmpty()
+     */
+    public Task(String description) throws IllegalArgumentException {
+        if (description.isEmpty()) {
+            throw new IllegalArgumentException("You can't do a non-existent task.");
+        }
+
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Returns the description of the task.
+     *
+     * @return a String representing the description of the task
+     */
+    public String saveRepr() {
+        return String.format("%d | %s", isDone ? 1 : 0, description);
+    }
+
+
+    /**
+     * Sets the status of the task.
+     *
+     * @param done a boolean that determines whether a task is done
+     */
+    public void setDone(boolean done) {
```

Comment:
I think the boolean name has to follow the naming standard.
```suggestion
    public void setDone(boolean isDone) {
```

Reply:
ooo, yea i must have missed out on that parameter. Thanks